src/diffpy/labpdfproc/tools.py

```diff
@@ -172,15 +172,31 @@ def set_wavelength(args):
     Raises
     ------
     ValueError
-        Raised when input wavelength is non-positive
-        or if input anode_type is not one of the known sources.
+        Raised if:
+        (1) neither is provided, and either mu*D needs to be looked up or
+            xtype is not the two-theta grid,
+        (2) both are provided,
+        (3) anode_type is not one of the known sources,
+        (4) wavelength is non-positive.
 
     Returns
     -------
     args : argparse.Namespace
         The updated arguments with the wavelength.
     """
-    if args.wavelength is None:
+    # first load values from config file
+    if args.wavelength is None and args.anode_type is None:
+        if not (args.mud is not None and args.xtype in ANGLEQUANTITIES):
```

Comment:
a comment might help readability here.  Whilst the logic can, in principle, be parsed from the code, double-negatives and ANDs make it hard, so a brief comment about the desired outcome could help here.  Normally, I like code to "self comment" and not put comments, and to simplify code to accomplish this, but here probably there is no way to simplify the logic correctly so a comment is ok.